File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long int lo = 0;
        long int hi = sqrt(c);
        while(lo <= hi){
            if(lo*lo + hi*hi == c)
                return true;
            else if(lo*lo + hi*hi > c)
                hi--;
            else
                lo++;
        }
        return false;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        first=0
        last=int(sqrt(c))
        if c<=2:
            return True
        
        while first<=last:
            k=(first*first) + (last*last) 
            if k==c:
                return True
            elif k<c:
                first=first+1
            else:
                last=last-1
        return False